src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentValidationServiceImpl.java

```diff
@@ -126,6 +138,78 @@ private boolean isExplanationValid(String explanation, CreateAccountingDocumentR
         return explanationIsValid;
     }
     
+    private boolean documentWasParsedWithoutErrors(AccountingXmlDocumentEntry accountingXmlDocument,
+            CreateAccountingDocumentReportItemDetail reportItemDetail) {
+        if (CollectionUtils.isNotEmpty(accountingXmlDocument.getValidationErrors())) {
+            for (ValidationEvent validationError : accountingXmlDocument.getValidationErrors()) {
+                reportItemDetail.appendErrorMessageToExistingErrorMessage(
+                        buildValidationErrorMessageForUser(validationError));
+            }
+            LOG.info("documentWasParsedWithoutErrors: Detected one or more parsing validation errors.");
+            return false;
+        } else {
+            return true;
+        }
+    }
+    
+    private String buildValidationErrorMessageForUser(ValidationEvent validationError) {
+        String subMessage = getErrorSubMessage(validationError);
+        int lineNumber = getErrorLineNumber(validationError);
+        if (lineNumber != -1) {
+            return formatMessageFromProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_XML_ADAPTER_ERROR,
+                    lineNumber, subMessage);
+        } else {
+            return subMessage;
+        }
+    }
+    
+    private String getErrorSubMessage(ValidationEvent validationError) {
+        String eventMessage = getErrorEventMessage(validationError);
```

Comment:
This function is getting rather long, can it be refactored into some smaller functions?

Reply:
I'll get this refactored into a few more smaller functions.